### PR TITLE
fix(Notion Node): Extract page url

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

Fix for extracting page url from urls of type
` https://www.notion.so/xxxxx?v=xxxxx&p=xxxxx&pm=s`

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1419/notion-node-database-page-param-cant-parse-pip-links

### CODE REVIEW

File: packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts [hunk 1]
```diff
@@ -916,6 +917,32 @@ export function extractPageId(page = '') {
...
+
+	if (page.value && typeof page.value === 'string') {
+		if (page.mode === 'id') {
+			pageId = page.value;
+		} else if (page.value.includes('p=')) {
+			// e.g https://www.notion.so/xxxxx?v=xxxxx&p=xxxxx&pm=s
+			pageId = page.value.split('p=')[1].split('&')[0];
+		} else {
+			// e.g https://www.notion.so/page_name-xxxxx
+			pageId = page.value.match(databasePageUrlValidationRegexp)?.[1] || '';
```
Comment (elsmr) [inline packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts:932]: nit: `??`

**Review by elsmr — DISMISSED** [review]

File: packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts [hunk 2]
```diff
@@ -916,6 +917,32 @@ export function extractPageId(page = '') {
...
+export function getPageId(this: IExecuteFunctions, i: number) {
+	const page = this.getNodeParameter('pageId', i, {}) as INodeParameterResourceLocator;
+	let pageId = '';
+
+	if (page.value && typeof page.value === 'string') {
+		if (page.mode === 'id') {
+			pageId = page.value;
+		} else if (page.value.includes('p=')) {
+			// e.g https://www.notion.so/xxxxx?v=xxxxx&p=xxxxx&pm=s
+			pageId = page.value.split('p=')[1].split('&')[0];
```
Comment (elsmr) [inline packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts:929]: nit: This works, but more robust with `URLSearchParams`

```suggestion
			pageId = new URLSearchParams(page.value).get('p')
```

File: packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts (hunk 1 above)
Comment (michael-radency) [reply packages/nodes-base/nodes/Notion/shared/GenericFunctions.ts:932]: this how it works in rlc when extracting value, taking match from regex

**Review by elsmr — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.68.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.68.0)

**Pull request merged**